medusa/helpers.py

```diff
@@ -1189,7 +1189,19 @@ def get_url(url, post_data=None, params=None, headers=None, timeout=30, session=
     hooks, cookies, verify, proxies = request_defaults(kwargs)
     method = u'POST' if post_data else u'GET'
 
+    # Get RequestPolice instance object.
+    rpolice = kwargs.pop(u'rpolice', None)
+
+    from helper.request_police import RequestPoliceException
+
     try:
+        if rpolice:
+            try:
+                [rpolice_check() for rpolice_check in rpolice.enabled_police_request_hooks]
+            except RequestPoliceException as e:
```

Comment:
In the init you can then do a `self.hook = ...` etc.   The required hooks for a service can then be passed as params to the PolicedSession's init.  As well as any user preferences such as API limit to reserve, etc.